src/query/functions/temporal/linear_regression.go

```diff
@@ -139,13 +139,17 @@ func (l linearRegressionNode) process(
 	return l.fn(slope, intercept)
 }
 
+func subSeconds(from int64, sub int64) float64 {
+	return float64(from-sub) / float64(time.Second)
+}
+
 // linearRegression performs a least-square linear regression analysis on the
 // provided datapoints. It returns the slope, and the intercept value at the
 // provided time.
 // Uses this algorithm: https://en.wikipedia.org/wiki/Simple_linear_regression.
 func linearRegression(
 	dps ts.Datapoints,
-	interceptTime time.Time,
+	interceptTime int64,
```

Comment:
nit: Can we use `xtime.UnixNano` instead of just pure `int64` here and other places? Feel like it will be safer/easier to grok/read too.  There's a lot of To/From helpers from that type too.

Reply:
Good call, changed all of these in the upcoming PR